tests/MongoDB.EntityFrameworkCore.FunctionalTests/Storage/TransactionTests.cs

```diff
@@ -526,39 +505,29 @@ public async Task Same_context_reads_its_own_writes_inside_explicit_transaction_
     }
 
     [Fact]
-    public void Disposing_explicit_transaction_without_commit_or_rollback_throws_and_does_not_commit()
+    public void Disposing_explicit_transaction_without_commit_or_rollback_does_not_commit()
     {
         var collection = database.CreateCollection<SimpleEntity>();
 
-        var ex = Assert.Throws<InvalidOperationException>(() =>
-        {
-            using var db = SingleEntityDbContext.Create(collection);
-            using var transaction = db.Database.BeginTransaction();
-            db.Entities.Add(new SimpleEntity { name = "pending" });
-            db.SaveChanges();
-            // Transaction goes out of scope without Commit/Rollback -> Dispose should throw
-        });
+        using var db = SingleEntityDbContext.Create(collection);
+        using var transaction = db.Database.BeginTransaction();
+        db.Entities.Add(new SimpleEntity { name = "pending" });
+        db.SaveChanges();
 
-        Assert.Contains("Dispose", ex.Message);
         using var check = SingleEntityDbContext.Create(collection);
         Assert.Empty(check.Entities);
     }
 
     [Fact]
-    public async Task Disposing_explicit_transaction_without_commit_or_rollback_throws_and_does_not_commit_async()
+    public async Task Disposing_explicit_transaction_without_commit_or_rollback_does_not_commit_async()
```

Comment:
This test used to ensure we also throw, but now we don't in order to be compatible with ExecutionStrategy.